### DISCUSSION TITLE
reconstruction for limited angles

### DISCUSSION THREAD

**Discussion by PeterPaang4** [astra-toolbox/astra-toolbox#466] [Q&A]
Hello, I hope I haven’t disturbed you. I would like to ask if this library has any functions or algorithms for limited-angle cone beam reconstruction. If so, can you tell me which function or algorithm it is?

**Comment by askorikov** [comment]
Hi! FDK algorithm has a "short scan" option for data acquired with less than 360 degrees rotation, if that's what you mean: https://astra-toolbox.com/docs/algs/FDK_CUDA.html

**Comment by PeterPaang4** [comment]
> Hi! FDK algorithm has a "short scan" option for data acquired with less than 360 degrees rotation, if that's what you mean: https://astra-toolbox.com/docs/algs/FDK_CUDA.html

Thank you, you are right. In addition, I need to de-artifact the image, and I need to modify the code myself. Do you know how and where I can see the source code and make modifications?

**Reply by askorikov** [reply]
If you mean the source code for the short scan correction, it's located here: https://github.com/astra-toolbox/astra-toolbox/blob/51561150bbc5d4ebc8f7a715e4c539542a5a5eb0/cuda/3d/fdk.cu#L192 You would need to build the Astra toolbox from source to test your modifications then. There's also an option to develop your algorithm on a higher level (outside of Astra) using the Tomosipo package as an Astra wrapper and PyTorch as a numerical computing library. See here for example: https://github.com/ahendriksen/ts_algorithms/blob/master/ts_algorithms/fdk.py

**Reply by PeterPaang4** [reply]
Hello, I hope I didn't bother you. I followed your instructions and used shortscan, but the error message Warning: Angular range (3.141593 rad) smaller than Parker weighting range (3.642828 rad). My data was collected only in the 180-degree range. Does shortscan require a range greater than 180?

**Reply by askorikov** [reply]
Hi! The range required for Parker weighting is pi + cone angle.

**Reply by PeterPaang4** [reply]
So do I need to modify the range of Parker weighting?

**Reply by PeterPaang4** [reply]
In addition,when using the fdk algorithm for reconstruction, it seems that only cubes can be constructed for volume geometry, and anisotropic volume geometry cannot be constructed. Do I have to modify it in the source code again?

**Reply by askorikov** [reply]
Depends on your objectives I guess. An smaller angular range will introduce additional artifacts, but if you're willing to accept that, you can modify the source code to apply the weighting nevertheless. I will add this to the backlog as a feature request.
